### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento

In [8]:
import pickle

In [9]:
with open('../../variaveis_preprocessadas/census.pkl', 'rb') as f:
    previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)

### Utilização do algoritmo Naive Bayes

In [10]:
from sklearn.naive_bayes import GaussianNB

gausian = GaussianNB()
gausian.fit(previsores_treinamento, classe_treinamento)
previsoes = gausian.predict(previsores_teste)

### Avaliação dos resultados do algoritmo Naive Bayes

In [11]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [12]:
score = accuracy_score(classe_teste, previsoes)
score

0.4767656090071648

In [13]:
confusion_matrix(classe_teste, previsoes)

array([[1172, 2521],
       [  35, 1157]], dtype=int64)

In [15]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

       <=50K       0.97      0.32      0.48      3693
        >50K       0.31      0.97      0.48      1192

    accuracy                           0.48      4885
   macro avg       0.64      0.64      0.48      4885
weighted avg       0.81      0.48      0.48      4885



### Resultados obtidos

O algoritmo possui 47% de acurácia, um resultado ruim considerando o fato que se ele escolhesse de maneira aleatória teria 50% de acerto. O algoritmo tem um rendimento muito baixo na detecção de pessoas que recebem mais de 50 mil reais, dos 97% detectados apenas 31% realmente recebem mais de 50 mil.